### PR TITLE
Parsing functions bugfixes

### PR DESCRIPTION
**Pull request by Aeristoka**
`handleSpecialGeneralAttacks` now checks, where appropriate for `to_hit != null`
Removed redundant checks in `handleSpecialMeleeAttacks` and `handleSpecialRangedAttacks` for `to_hit != null`, as that's handled in the function calling logic
Re-ordered some parts of `handleSpecialSpells` and `handleSpecialHealingSpells` so that features which work on ALL damage types now have their chance to do so.

### CODE REVIEW

File: src/dndbeyond/content-scripts/character.js [hunk 1]
```diff
@@ -326,8 +326,7 @@ function handleSpecialMeleeAttacks(damages=[], damage_types=[], properties, sett
 
     // Feats
     // Great Weapon Master Feat
-    if (to_hit !== null && 
-        character.getSetting("great-weapon-master", false) &&
+    if (character.getSetting("great-weapon-master", false) &&
```
Comment (kakaroto) [inline src/dndbeyond/content-scripts/character.js:329]: why did you remove the check for to_hit in here?

File: src/dndbeyond/content-scripts/character.js [hunk 2]
```diff
@@ -513,8 +514,7 @@ function handleSpecialWeaponAttacks(damages=[], damage_types=[], properties, set
 
     if (character.hasClass("Paladin")) {
         // Paladin: Sacred Weapon
-        if (to_hit !== null && 
-            character.getSetting("paladin-sacred-weapon", false)) {
+        if (character.getSetting("paladin-sacred-weapon", false)) {
```
Comment (kakaroto) [inline src/dndbeyond/content-scripts/character.js:517]: same question about removing check for to_hit

**Review by kakaroto — COMMENTED** [review]
brain = dead. will leave this comments and retry to review some other day.

File: src/dndbeyond/content-scripts/character.js (hunk 1 above)
Comment (Aeristoka) [reply src/dndbeyond/content-scripts/character.js:329]: There is no condition under which there isn't a `to_hit` on this now in the function call

File: src/dndbeyond/content-scripts/character.js (hunk 2 above)
Comment (Aeristoka) [reply src/dndbeyond/content-scripts/character.js:517]: There is no condition under which there isn't a `to_hit` on this now in the function call

File: src/dndbeyond/content-scripts/character.js [hunk 3]
```diff
@@ -1131,36 +1129,55 @@ function handleSpecialSpells(spell_name, damages=[], damage_types=[], {spell_sou
...
-        for (let i = 0; i < damages.length; i++){
-            if (damage_types[i] === "Healing") {
-                damages.push("1d8");
-                damage_types.push("Enhanced Bond Healing");
-                break;
+    // Supreme Healing must ALWAYS be at the end, at it maxes all healing dice
+    if (character.hasClass("Cleric")) {
+        if (character.hasClassFeature("Supreme Healing")) {
+            for (let i = 0; i < damages.length; i++) {
+                if (damage_types[i] !== "Healing") continue;
```
Comment (kakaroto) [inline src/dndbeyond/content-scripts/character.js:1177]: Shouldn't this check for `.includes("Healing")` because of things like "Enhanced Bond Healing" or "Alchemical Savant Healing" ?

File: src/dndbeyond/content-scripts/character.js [hunk 4]
```diff
@@ -1249,6 +1258,11 @@ function rollSpell(force_display = false, force_to_hit_only = false, force_damag
         if (healing_modifiers.length > 0) {
             handleSpecialHealingSpells(spell_name, damages, damage_types, {spell_level: level, spell_source, castas});
         }
+        else {
+            to_hit = handleSpecialGeneralAttacks(damages, damage_types, properties, settings_to_change, {to_hit, spell_name, spell_level: level});
+        
+            handleSpecialSpells(spell_name, damages, damage_types, {spell_level: level, spell_source, castas});
+        }
```
Comment (kakaroto) [inline src/dndbeyond/content-scripts/character.js:1265]: Why is `handleSpecialSpells` inside the `else` so it's only for non healing spells ? I think it should be outside the else (even if there is nothing that would apply for it in non-healing spells)
Also did the condition that made you call `handleSpecialGeneralAttacks` for non-hunters-mark not apply anymore ?

File: src/dndbeyond/content-scripts/character.js [hunk 5]
```diff
@@ -1248,6 +1257,11 @@ function rollSpell(force_display = false, force_to_hit_only = false, force_damag
         if (healing_modifiers.length > 0) {
             handleSpecialHealingSpells(spell_name, damages, damage_types, {spell_level: level, spell_source, castas});
         }
+        else {
```
[the remark this answers is not in the document]
Comment (kakaroto) [reply src/dndbeyond/content-scripts/character.js:1261]: Yep, it's fine after you explained to me on Discord the reason for using the `else`. I just needed to understand it.

[the remark this answers is not in the document]
Comment (Aeristoka) [reply src/dndbeyond/content-scripts/character.js:1261]: But I already changed it 😭

File: src/dndbeyond/content-scripts/character.js (hunk 3 above)
Comment (Aeristoka) [reply src/dndbeyond/content-scripts/character.js]: Done

**Comment by kakaroto** [comment]
oh yeah, much cool, much nice, much thanks!
